apollo-integration/src/test/java/com/apollographql/apollo/ApolloWatcherTest.java

```diff
@@ -104,6 +111,50 @@ public void testQueryWatcherUpdated_SameQuery_DifferentResults() throws IOExcept
   }
 
   @Test
+  public void testQueryWatcherUpdated_Store_write() throws IOException, InterruptedException,
+      TimeoutException, ApolloException {
+    final NamedCountDownLatch firstResponseLatch = new NamedCountDownLatch("firstResponseLatch", 1);
+    final NamedCountDownLatch secondResponseLatch = new NamedCountDownLatch("secondResponseLatch", 2);
+
+    EpisodeHeroNameQuery query = EpisodeHeroNameQuery.builder().episode(Episode.EMPIRE).build();
+    server.enqueue(mockResponse("EpisodeHeroNameResponseWithId.json"));
+
+    ApolloQueryWatcher<EpisodeHeroNameQuery.Data> watcher = apolloClient.query(query).watcher();
+    watcher.enqueueAndWatch(
+        new ApolloCall.Callback<EpisodeHeroNameQuery.Data>() {
+          @Override public void onResponse(@Nonnull Response<EpisodeHeroNameQuery.Data> response) {
+            if (secondResponseLatch.getCount() == 2) {
+              assertThat(response.data().hero().name()).isEqualTo("R2-D2");
```

Comment:
I see... I just copy/pasted the test above so we might need to fix some of them. We still need to check that so my favorite solution would be to have the callback run in the main thread. I know how to do that in RxJava, not sure we want to use RxJava in these files though. Any idea what would be the best solution here ?